src/cryptography/hazmat/bindings/openssl/ssl.py

```diff
@@ -7,6 +7,14 @@
 INCLUDES = """
 #include <openssl/ssl.h>
 
+/* LibreSSL has removed support for compression, and with it the
+ * COMP_METHOD use in ssl.h. This is a hack to make the function types
+ * in this code match those in ssl.h.
+ */
+#ifdef LIBRESSL_VERSION_NUMBER
+#define COMP_METHOD void
+#endif
```

Comment:
Yeah, although that starts to bleed into #1675 :grinning: 

I'll move this so we can merge today.